bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/GenericItem.java

```diff
@@ -340,7 +340,7 @@ public boolean equals(@Nullable Object obj) {
 
     @Override
     public boolean hasTag(String tag) {
-        return (tags.contains(tag));
+        return tags.stream().map(String::toLowerCase).collect(Collectors.toList()).contains(tag.toLowerCase());
```

Comment:
I know that `tag` is marked as `NonNull`, but since this is part of the public API, we should make sure to not accidentally introduce a possible NPE here.

So what about using `return tags.stream().anyMatch(t -> t.equalsIgnoreCase(tag));`?

Reply:
Yes, looks simpler and cleaner.